### PR TITLE
Add Cloud Network to Circuit Termination

### PR DESCRIPTION
**Pull request by smk4664**
Add Cloud Network as an option for Circuit Terminations, this is for the model and REST API only.

# Partial work towards #5716 
# What's Changed

This change will add Cloud Networks as an option for Circuit Terminations in the Model and REST API. 
# Screenshots

N/A
# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/circuits/models.py [hunk 1]
```diff
@@ -215,22 +215,35 @@ class CircuitTermination(PrimaryModel, PathEndpoint, CableTermination):
     xconnect_id = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True, verbose_name="Cross-connect ID")
     pp_info = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True, verbose_name="Patch panel/port(s)")
     description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
+    cloud_network = models.ForeignKey(
+        to="cloud.CloudNetwork",
+        on_delete=models.PROTECT,
+        related_name="circuit_terminations",
+        blank=True,
+        null=True,
+    )
```
Comment (glennmatthews) [inline nautobot/circuits/models.py:224]: Maybe move this up so it's adjacent to the `location` and `provider_network` field declarations?

File: nautobot/circuits/tests/test_filters.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ class CircuitTerminationTestCase(FilterTestCases.FilterTestCase):
     generic_filter_tests = (
         ["circuit", "circuit__cid"],
         ["circuit", "circuit__id"],
+        ["cloud_network", "cloud_network__name"],
+        ["cloud_network", "cloud_network__id"],
```
Comment (glennmatthews) [inline nautobot/circuits/tests/test_filters.py:133]: Should be some additions to the CircuitTestCase as well, right?

File: nautobot/circuits/tests/test_models.py [hunk 3]
```diff
@@ -28,11 +29,19 @@ def setUpTestData(cls):
...
+            cloud_type=cloud_type,
+            name="Cloud Network 1",
+        )
+
+    def test_location_or_provider_network_or_cloud_network_are_required(self):
         ct = CircuitTermination(circuit=self.circuit, term_side=CircuitTerminationSideChoices.SIDE_A)
         with self.assertRaises(ValidationError) as cm:
             ct.validated_save()
-        self.assertIn("must attach to either a location or a provider network", str(cm.exception))
+        self.assertIn("must attach to one of a location, a provider network or a cloud network", str(cm.exception))
```
Comment (HanlinMiao) [inline nautobot/circuits/tests/test_models.py:44]:
```suggestion
        self.assertIn("must attach to a location, a provider network or a cloud network", str(cm.exception))
```

File: nautobot/docs/user-guide/core-data-model/circuits/circuittermination.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
 
-Each circuit termination is attached to either a location or a provider network. Location terminations may optionally be connected via a cable to a specific device interface or port within that location. Each termination must be assigned a port speed, and can optionally be assigned an upstream speed if it differs from the downstream speed (a common scenario with e.g. DOCSIS cable modems). Fields are also available to track cross-connect and patch panel details.
+Each circuit termination is attached to either a location, a provider network or a cloud network. Location terminations may optionally be connected via a cable to a specific device interface or port within that location. Each termination must be assigned a port speed, and can optionally be assigned an upstream speed if it differs from the downstream speed (a common scenario with e.g. DOCSIS cable modems). Fields are also available to track cross-connect and patch panel details.
```
Comment (HanlinMiao) [inline nautobot/docs/user-guide/core-data-model/circuits/circuittermination.md:5]:
```suggestion
Each circuit termination is attached to a location, a provider network or a cloud network. Location terminations may optionally be connected via a cable to a specific device interface or port within that location. Each termination must be assigned a port speed, and can optionally be assigned an upstream speed if it differs from the downstream speed (a common scenario with e.g. DOCSIS cable modems). Fields are also available to track cross-connect and patch panel details.
```

**Review by HanlinMiao — APPROVED** [review]

**Comment by HanlinMiao** [comment]
There is a fix for the current unit test failure here https://github.com/nautobot/nautobot/pull/5881

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**